### ISSUE TITLE
Should we make it possible to put the logo on the right?

### ISSUE THREAD

**Issue by allisio** [sam-barr/paleofetch#44]
I did [a side-by-side](https://i.imgur.com/p6DTb6R.png) of my current configuration and what I wish it looked like, and I thought maybe other folks would like to have the logo on the right (either justified or "as flush as possible" with the data like I have it in the demo).

The problem is this would be a fairly invasive change to the code. We wouldn't know how to position things until the end of data acquisition when we know the length of the longest line. A somewhat clean approach might be to gate all the right-alignment code behind an `#ifdef RIGHT`, and then instead of straight `printf()` in `main()`, we print to a string and then do all the magic on that *iff* we're trying to right-align. But then we're left dealing with the ANSI escape sequences contributing to line lengths.

It could get hairy, but I think this would be a nifty feature that sets us apart from `neofetch` (not that the absurd speed improvement is anything to scoff at).

**Comment by ss7m** [comment]
Dealing with the escape sequences wouldn't be toooo hard, since (aside from `get_title`) you can just subtract `sizeof(COLOR)` and `sizeof("\e[0m")`. Anything more complicated than that would likely involve having to parse ANSI escape sequences, which sounds... not fun. 

And we already calculate the displayed length of `get_title` for `get_bar` anyways

**Comment by ss7m** [comment]
I believe `snprintf` returns the number of characters written, perhaps we can take advantage of that

**Comment by ss7m** [comment]
I'm working on a solution to finding the maximum length right now

**Comment by ss7m** [comment]
Check out my latest commit on the allignemt branch I just created. What do you make of that?

**Comment by allisio** [comment]
Hm... I think that's touching an awful lot of things, but maybe it's still the right way to do it.

I was thinking something more like [this](https://github.com/allisio/paleofetch/commit/7671539e70fb72068ff3823c2e4aef5929280dc2), where we just redirect where we're printing to. At the end, we have what we would've printed in `output` and can finagle it as necessary to reposition the logo.

That's the hard part, but I think you were right about just being able to subtract the lengths of some static strings in order to get a handle on how to start aligning.

**Comment by ss7m** [comment]
Yeah I became less and less of a fan of my solution as I wrote it. I'm interested to see what you can come up with

**Comment by ss7m** [comment]
Just a thought, but any UTF8 weirdness in any of the output would further complicate this
